nsxt/resource_nsxt_compute_manager.go

```diff
@@ -0,0 +1,550 @@
+										Optional:    true,
+									},
+									"thumbprint": {
+										Type:        schema.TypeString,
+										Description: "Thumbprint of the login server",
+										Optional:    true,
+									},
+									"username": {
+										Type:        schema.TypeString,
+										Description: "The username for login",
+										Optional:    true,
+									},
+								},
+							},
+						},
+						"verifiable_asymmetric_login_credential": {
+							Type:        schema.TypeList,
+							Description: "A verifiable asymmetric login credential",
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"asymmetric_credential": {
+										Type:        schema.TypeString,
+										Description: "Asymmetric login credential",
+										Optional:    true,
+									},
+									"credential_key": {
+										Type:        schema.TypeString,
+										Description: "Credential key",
+										Optional:    true,
+									},
+									"credential_verifier": {
+										Type:        schema.TypeString,
+										Description: "Credential verifier",
+										Optional:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"extension_certificate": {
+				Type:        schema.TypeList,
+				Description: "Specifies certificate for compute manager extension",
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"pem_encoded": {
+							Type:        schema.TypeString,
+							Description: "PEM encoded certificate data",
+							Required:    true,
+						},
+						"private_key": {
+							Type:        schema.TypeString,
+							Description: "Private key of certificate",
+							Required:    true,
+						},
+					},
+				},
+			},
+			"multi_nsx": {
+				Type:        schema.TypeBool,
+				Description: "Specifies whether multi nsx feature is enabled for compute manager",
+				Optional:    true,
+			},
+			"origin_properties": {
+				Type:        schema.TypeList,
+				Description: "Key-Value map of additional specific properties of compute manager",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:        schema.TypeString,
+							Description: "Key",
+							Required:    true,
+						},
+						"value": {
+							Type:        schema.TypeString,
+							Description: "Value",
+							Required:    true,
+						},
+					},
+				},
+			},
+			"origin_type": {
+				Type:        schema.TypeString,
+				Description: "Compute manager type like vCenter",
+				Required:    true,
+			},
+			"reverse_proxy_https_port": {
+				Type:        schema.TypeInt,
+				Description: "Proxy https port of compute manager",
+				Optional:    true,
+			},
+			"server": {
+				Type:        schema.TypeString,
+				Description: "IP address or hostname of compute manager",
+				Required:    true,
+			},
+			"set_as_oidc_provider": {
```

Comment:
Default value here?

Reply:
I've added defaults to all the attributes which have default values in NSX. But I'm not sure I understand this: if TF assigns a default value, wouldn't it clutter the state file with stuff that wasn't assigned by the user?

Reply:
If NSX assigns default, it will make its way to the state anyway.